torch/utils/tensorboard/writer.py

```diff
@@ -0,0 +1,731 @@
+        """
+        self.event_writer.reopen()
+
+
+class SummaryWriter(object):
+    """Writes entries directly to event files to be consumed by TensorBoard.
+
+    The `SummaryWriter` class provides a high-level API to create an event file
+    in a given directory and add summaries and events to it. The class updates the
+    file contents asynchronously. This allows a training program to call methods
+    to add data to the file directly from the training loop, without slowing down
+    training.
+    """
+
+    def __init__(self, log_dir=None, comment='', **kwargs):
+        """Creates a `SummaryWriter` which uses a `FileWriter` internally to
+        write out events and summaries to the event file.
+
+        Args:
+            log_dir (string): save location, default is: runs/**CURRENT_DATETIME_HOSTNAME**, which changes after each
+              run. Use hierarchical folder structure to compare between runs easily. e.g. 'runs/exp1', 'runs/exp2'
+            comment (string): comment that appends to the default ``log_dir``. If ``log_dir`` is assigned,
+              this argument will no effect.
+            purge_step (int):
+              When logging crashes at step :math:`T+X` and restarts at step :math:`T`, any events
+              whose global_step larger or equal to :math:`T` will be purged and hidden from TensorBoard.
+              Note that the resumed experiment and crashed experiment should have the same ``log_dir``.
+            filename_suffix (string):
+              Every event file's name is suffixed with suffix. Example: ``SummaryWriter(filename_suffix='.123')``
+              More details on event filename construction in
+              tensorboard.summary.writer.event_file_writer.EventFileWriter.
+            kwargs: extra keyword arguments for FileWriter (e.g. 'flush_secs'
+              controls how often to flush pending events). For more arguments
+              please refer to docs for 'tf.summary.FileWriter'.
+        """
+        if not log_dir:
+            import socket
+            from datetime import datetime
+            current_time = datetime.now().strftime('%b%d_%H-%M-%S')
+            log_dir = os.path.join(
+                'runs', current_time + '_' + socket.gethostname() + comment)
+        self.log_dir = log_dir
+        self.kwargs = kwargs
+
+        # Initialize the file writers, but they can be cleared out on close
+        # and recreated later as needed.
+        self.file_writer = self.all_writers = None
+        self.get_file_writer()
+
+        # Create default bins for histograms, see generate_testdata.py in tensorflow/tensorboard
+        v = 1E-12
+        buckets = []
+        neg_buckets = []
+        while v < 1E20:
+            buckets.append(v)
+            neg_buckets.append(-v)
+            v *= 1.1
+        self.default_bins = neg_buckets[::-1] + [0] + buckets
+        self.scalar_dict = {}
+
+    def __append_to_scalar_dict(self, tag, scalar_value, global_step,
```

Comment:
Is there any particular reason this has a double underscore leading prefix? General idiomatic Python eschews use of double underscore, unless you have a good reason. If you just want private, do a single underscore.

Reply:
Changing.